src/validation/__tests__/ProhibitDeprecatedFieldsRule-test.js

```diff
@@ -0,0 +1,135 @@
+// @flow strict
+
+import { describe, it } from 'mocha';
+
+import { buildSchema } from '../../utilities/buildASTSchema';
+
+import { ProhibitDeprecatedFieldsRule } from '../rules/optional/ProhibitDeprecatedFieldsRule';
+
+import { expectValidationErrorsWithSchema } from './harness';
+
+const schema = buildSchema(`
+  enum EnumType {
+    NORMAL_VALUE
+    DEPRECATED_VALUE @deprecated(reason: "Some enum reason.")
+    DEPRECATED_VALUE_WITH_NO_REASON @deprecated
+  }
+
+  type Query {
+    normalField(enumArg: [EnumType]): String
+    deprecatedField: String @deprecated(reason: "Some field reason.")
+    deprecatedFieldWithNoReason: String @deprecated
+  }
+`);
+
+function expectErrors(queryStr) {
+  return expectValidationErrorsWithSchema(
+    schema,
+    ProhibitDeprecatedFieldsRule,
+    queryStr,
+  );
+}
+
+function expectValid(queryStr) {
+  expectErrors(queryStr).to.deep.equal([]);
+}
+
+describe('Validate: prohibit deprecated fields', () => {
+  it('ignores fields and enum values that are not deprecated', () => {
+    expectValid(`
+      {
+        normalField(enumArg: [NORMAL_VALUE])
+      }
+    `);
+  });
+
+  it('ignores unknown fields and enum values', () => {
+    expectValid(`
+      fragment UnknownFragment on UnknownType {
+        unknownField(unknownArg: UNKNOWN_VALUE)
+      }
+    `);
```

Comment:
You not testing unknown field on known type combination.
That's why coverage showed that branch as uncovered in optional chaining.

Reply:
the same problem with enum value you always check the only left side of `&&`.